core/src/kmx/kmx_plus.cpp

```diff
@@ -9,24 +9,34 @@
 #include <kmx/kmx_plus.h>
 #include <kmx/kmx_xstring.h>
 
-#if KMXPLUS_DEBUG
-#include <iostream>
-#include <stdio.h>
-#endif
+#include "kmx_processevent.h" // for debug functions
 
 #include <assert.h>
 
 namespace km {
 namespace kbp {
 namespace kmx {
 
-#if KMXPLUS_DEBUG
-#define debug_println(x) std::cerr << __FILE__ << ":" << __LINE__ << ": " << (x) << std::endl;
-#define KMXPLUS_PRINTF(x) printf x
-#else
-#define debug_println(x)
-#define KMXPLUS_PRINTF(x)
-#endif
+/**
+ * @brief convert a section name to string, for debugging
+ * @param ident the hex dword
+ * @param buf buffer, must be size 5
+ * @return const pointer to buffer
+ */
+inline const char *
+section_name_to_str(KMX_DWORD ident, char *buf) {
```

Comment:
I'm just having a bit of fun here, but another way of skinning this would be:

```c++
#define _DEBUG_IDENT_SAFE(x,b) (((x)>>b)&0x7F)<0x20?'?':(((x)>>b)&0x7F)
#define DEBUG_IDENT(x) \
  _DEBUG_IDENT_SAFE(x,0), \
  _DEBUG_IDENT_SAFE(x,8), \
  _DEBUG_IDENT_SAFE(x,16), \
  _DEBUG_IDENT_SAFE(x,24)

  ...
  DebugLog("%c%c%c%c", DEBUG_IDENT(ident));
```

The primary advantage is it avoids loops (even if they are unwound), and reduces the debug statements to a single line, which reduces noise in the source.

Reply:
done